src/test/java/io/cryostat/rules/RuleMatcherTest.java

```diff
@@ -0,0 +1,256 @@
+
+@ExtendWith(MockitoExtension.class)
+class RuleMatcherTest {
+
+    RuleMatcher ruleMatcher;
+    @Mock ServiceRef serviceRef;
+
+    URI serviceUri;
+    String alias;
+    Map<String, String> labels;
+    Map<String, String> platformAnnotations;
+    Map<AnnotationKey, String> cryostatAnnotations;
+
+    @BeforeEach
+    void setup() throws Exception {
+        this.ruleMatcher = new RuleMatcher();
+
+        this.serviceUri = new URI("service:jmx:rmi:///jndi/rmi://cryostat:9091/jmxrmi");
+        this.alias = "someAlias";
+        this.labels = Map.of("label1", "someLabel");
+        this.platformAnnotations = Map.of("annotation1", "someAnnotation");
+        this.cryostatAnnotations = Map.of(AnnotationKey.JAVA_MAIN, "io.cryostat.Cryostat");
+
+        Mockito.when(serviceRef.getServiceUri()).thenReturn(this.serviceUri);
+        Mockito.when(serviceRef.getAlias()).thenReturn(Optional.of(this.alias));
+        Mockito.when(serviceRef.getLabels()).thenReturn(this.labels);
+        Mockito.when(serviceRef.getPlatformAnnotations()).thenReturn(this.platformAnnotations);
+        Mockito.when(serviceRef.getCryostatAnnotations()).thenReturn(this.cryostatAnnotations);
+    }
+
+    @Nested
+    class BindingsContents {
+        Bindings bindings;
+
+        @BeforeEach
+        void setup() {
+            this.bindings = ruleMatcher.createBindings(serviceRef);
+        }
+
+        @Test
+        void bindingsShouldContainTargetReference() {
+            MatcherAssert.assertThat(bindings, Matchers.hasKey("target"));
+            MatcherAssert.assertThat(bindings.size(), Matchers.equalTo(1));
+        }
+
+        @Test
+        void targetShouldHaveExpectedKeys() {
+            Set<String> keys = ((Map<String, Object>) bindings.get("target")).keySet();
+            MatcherAssert.assertThat(
+                    keys, Matchers.equalTo(Set.of("connectUrl", "alias", "labels", "annotations")));
+        }
+
+        @Test
+        void targetShouldHaveServiceUriAsUri() {
+            URI uri = (URI) ((Map<String, Object>) bindings.get("target")).get("connectUrl");
+            MatcherAssert.assertThat(uri, Matchers.equalTo(RuleMatcherTest.this.serviceUri));
+        }
+
+        @Test
+        void targetShouldHaveAliasAsString() {
+            String alias = (String) (((Map<String, Object>) bindings.get("target")).get("alias"));
+            MatcherAssert.assertThat(alias, Matchers.equalTo(RuleMatcherTest.this.alias));
+        }
+
+        @Test
+        void targetShouldHaveLabels() {
+            Map<String, String> labels =
+                    (Map<String, String>)
+                            (((Map<String, Object>) bindings.get("target")).get("labels"));
+            MatcherAssert.assertThat(labels, Matchers.equalTo(RuleMatcherTest.this.labels));
+        }
+
+        @Test
+        void annotationsShouldHaveExpectedFields() {
+            Bindings bindings = ruleMatcher.createBindings(serviceRef);
```

Comment:
I think this is unnecessary because it's just a duplicate of the class field `bindings`.